File: examples/runnable/01-hello-temporal/go/greeting.go
// Hello Temporal — the Go equivalent of the Java GreetingWorkflow lab.
package main

import (
	"context"
	"time"

	"go.temporal.io/sdk/workflow"
)

const TaskQueue = "hello-temporal"

// ComposeGreeting is an Activity (first arg is context.Context).
func ComposeGreeting(ctx context.Context, name string) (string, error) {
	return "Hello, " + name + " from a Temporal Activity", nil
}

// GreetingWorkflow calls the Activity and returns its result.
func GreetingWorkflow(ctx workflow.Context, name string) (string, error) {
	ctx = workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
		StartToCloseTimeout: 10 * time.Second,
	})
	var greeting string
	err := workflow.ExecuteActivity(ctx, ComposeGreeting, name).Get(ctx, &greeting)
	return greeting, err
}

File: examples/runnable/01-hello-temporal/python/greeting.py
"""Hello Temporal — the Python equivalent of the Java GreetingWorkflow lab."""

from datetime import timedelta

from temporalio import activity, workflow

TASK_QUEUE = "hello-temporal"


@activity.defn
async def compose_greeting(name: str) -> str:
    return f"Hello, {name} from a Temporal Activity"


@workflow.defn
class GreetingWorkflow:
    @workflow.run
    async def greet(self, name: str) -> str:
        return await workflow.execute_activity(
            compose_greeting, name, start_to_close_timeout=timedelta(seconds=10)
        )
